### PR TITLE
ACPI platform source validation with mock acpi

### PR DESCRIPTION
**Pull request by vimalk78**
This PR
- adds ability to mock acpi power source.
- updates compose files to run mock acpi , and adds dashboards for validation

sample dashboard
image
image

### CODE REVIEW

File: manifests/compose/compose.mock-acpi.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      - ALL
+
+  # OVERRIDE default dashboard
+  grafana:
+    environment:
+      GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH: "/var/lib/grafana/dashboards/dev/dashboard.json"
+
+    volumes:
+      - type: bind
+        source: ./grafana/dashboards/dev/
```
Comment (vprashar2929) [inline manifests/compose/compose.mock-acpi.yaml:63]: Shouldn't this point to `./grafana/dashboards/mock-acpi/` ?

File: manifests/compose/compose.mock-acpi.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        target: /var/lib/grafana/dashboards/dev
```
Comment (vprashar2929) [inline manifests/compose/compose.mock-acpi.yaml:64]: same here as well `/var/lib/grafana/dashboards/mock-acpi/`

**Comment by SamYuan1990** [comment]
may I know if https://kwok.sigs.k8s.io/docs/user/resource-usage-configuration/ can help us to mock the pod usage for testing? or this PR is more focusing on ACPI as hardware part?

File: pkg/sensors/platform/source/acpi.go [hunk 3]
```diff
@@ -52,7 +52,11 @@ type ACPI struct {
 	powerPath     string
 }
 
-func NewACPIPowerMeter() *ACPI {
+func NewACPIPowerMeter(mockpath string) *ACPI {
+	if mockpath != "" {
+		klog.V(3).Infof("using user provided ACPI power path: %s", mockpath)
```
Comment (sthaha) [inline pkg/sensors/platform/source/acpi.go:57]:
```suggestion
		klog.Infof("using user provided ACPI power path: %s", mockpath)
```

**Comment by vimalk78** [comment]
in the mock-acpi-data chart, divide by 10^6

File: pkg/sensors/platform/source/acpi.go (hunk 3 above)
Comment (vimalk78) [reply pkg/sensors/platform/source/acpi.go]: updated

File: manifests/compose/compose.mock-acpi.yaml (hunk 2 above)
Comment (vimalk78) [reply manifests/compose/compose.mock-acpi.yaml]: updated

**Comment by vimalk78** [comment]
sample dashboard
image
image

**Comment by vimalk78** [comment]
> or this PR is more focusing on ACPI as hardware part?

@SamYuan1990 yes, this PR is for mocking the ACPI power source only

File: manifests/compose/compose.mock-acpi.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        # NOTE: uncomment to wait for estimator container
+        # echo waiting for estimator socket to be ready;
+        # until [[ -e /tmp/estimator.sock ]]; do
+        #   echo " ... waiting for socket";
+        #   sleep 1;
+        # done;
+        set -x;
+        /usr/bin/kepler \
+          -address "0.0.0.0:8888" \
+          -v "3" \
```
Comment (jichenjc) [inline manifests/compose/compose.mock-acpi.yaml:48]: usually it's 1 for normal test and 5 for debug
what's the reason to set 3 here?

Comment (vimalk78) [reply manifests/compose/compose.mock-acpi.yaml:48]: no specific reason, this not hard coded in code, we can change it when we run the test.

**Comment by vimalk78** [comment]
@marceloamaral @rootfs PTAL, thanks

**Pull request merged**